src/data/set/finite.lean

```diff
@@ -215,6 +215,8 @@ fintype.card_of_subsingleton _
 lemma subsingleton.finite {s : set α} (h : s.subsingleton) : finite s :=
 h.induction_on finite_empty finite_singleton
 
+lemma to_finset_singleton (a : α) : ({a} : set α).to_finset = {a} := rfl
```

Comment:
I don't think this lemma, and the two new ones below, should be in this file, they should be in `data/fintype/basic` where `to_finset` is defined. For instance, if I just import that file, I should have access to this lemma whether or not `set.finite` is in my local context.

Reply:
There are only a handle of `to_finset` lemmas in `data/fintype/basic`, and most of the `set.to_finset_*` lemmas seem to be in `data/set/finite`.  Sorting through which should go where seems like it might be a project beyond this PR.

Reply:
As far as I can see, there are only five lemmas there which don't involve `set.finite.to_finset`, one of which is a duplicate of a lemma already in `data/fintype/basic`! And there are at least 10 lemmas of this form in `data/fintype/basic` already, so there is precedent for this change. Nonetheless if you don't want to make this PR take longer, I'm fine with a TODO above this lemma.

Reply:
Ok, I'll create a github issue about `to_finset` lemmas.

Reply:
Created #11177